loopy/kernel/function_interface.py

```diff
@@ -348,6 +348,8 @@ def __init__(self,
         try:
             hash(arg_id_to_dtype)
         except TypeError:
+            if arg_id_to_dtype is None:
+                arg_id_to_dtype = {}
```

Comment:
I'm not sure whether this is the intended behavior here (i.e., whether `arg_id_to_dtype` should just be set to `None`). This problem seems to have come in with https://github.com/inducer/loopy/commit/70cc100753981e80aac60f93aa5c2ed62230a1e7#diff-c2536267212ab595d8fbafdc40257bf749f171cc1e23c6c5113fcf2e41546b02

Same below.

Reply:
I think this can just be `assert arg_id_to_dtype is not None` (since `None` is hashable, we should not end up in this branch).

Reply:
Changed in [ba93a2f](https://github.com/inducer/loopy/pull/884/commits/ba93a2f101b0ae665a2ab425d7fc43b26c1bf5ae)